### PR TITLE
Config for timeouts and delays in probes

### PR DESCRIPTION
**Pull request by jan-kantert**
Proposed change to implement #11453

### CODE REVIEW

**Review by adleong — COMMENTED** [review]
Thanks @jan-kantert!

The default probe timeouts are 1s, so that's the current behavior.  Changing the defaults from 1s to 30s and 5s respectively feels like a big change.  I'm not sure how we would determine if these defaults are better.

What do you think about leaving the defaults as the current value of 1s?  At least this change will allow users to configure these timeouts explicitly without changing the default behavior.

**Comment by jan-kantert** [comment]
> Thanks @jan-kantert!
> 
> The default probe timeouts are 1s, so that's the current behavior. Changing the defaults from 1s to 30s and 5s respectively feels like a big change. I'm not sure how we would determine if these defaults are better.
> 
> What do you think about leaving the defaults as the current value of 1s? At least this change will allow users to configure these timeouts explicitly without changing the default behavior.

Sure we can do that. However, those small timeouts cause instability on clusters with high load. I personally would still at least increase the default for the livenessProbe (as Kubernetes recommends to have a longer timeout for your livenessProbe than your readinessProbe). This timeout also has been introduced more or less silently with kubernetes 1.20 (https://kubernetes.io/blog/2020/12/08/kubernetes-1-20-release-announcement/#exec-probe-timeout-handling).

**Review by alpeb — COMMENTED** [review]
Thanks @jan-kantert :+1: 
Please complete the `Proxy` struct in `./pkg/charts/linkerd2/values.go` with these new entries for the `linkerd install` CLI to be able to render them.
Also, please run `./bin/helm-docs` for the comments added into the `values.yaml` file to be picked up in the chart's README.md file, and `go test ./... -update` to update the unit test golden files.

**Comment by adleong** [comment]
Hi @jan-kantert, just checking in to see if you have time to make the above changes?

**Comment by mateiidavid** [comment]
Hey @jan-kantert, just to give you an update: I pushed the requested changes to your branch. I did not change anything in the implementation (but I did modify the file to resolve a merge conflict) -- most of my work has been to fix the tests and get this reviewable.

Your effort is super appreciated! I think due to the nature of the issue itself (that you have described in #11453) we felt that it would be better to get this fix in soon so we can get it tested asap. 

The contribution is solely yours :)

File: charts/partials/templates/_proxy.tpl [hunk 1]
```diff
@@ -159,7 +159,8 @@ livenessProbe:
   httpGet:
     path: /live
     port: {{.Values.proxy.ports.admin}}
-  initialDelaySeconds: 10
+  initialDelaySeconds: {{.Values.proxy.livenessProbe.initialDelaySeconds | default 10}}
+  timeoutSeconds: {{.Values.proxy.livenessProbe.timeoutSeconds | default 1}}
```
Comment (alpeb) [inline charts/partials/templates/_proxy.tpl:163]: I would get rid of the `default` statements, as we already have defaults in the `values.yaml` file, and to be consistent with the rest of the file (see `startupProbe`).

File: charts/linkerd-control-plane/values.yaml [hunk 2]
```diff
@@ -198,6 +198,14 @@ proxy:
   # "all-unauthenticated", "cluster-authenticated", "cluster-unauthenticated", "deny"
   # @default -- "all-unauthenticated"
   defaultInboundPolicy: "all-unauthenticated"
+  # -- LivenessProbe timeout and delay configuration
+  livenessProbe:
+    initialDelaySeconds: 10
```
Comment (alpeb) [inline charts/linkerd-control-plane/values.yaml:203]: So the default here used to be zero? Setting the new default to 10 is going to significantly delay the startup time for pods. How about leaving the default as it was? Pathological cases would still have an escape hatch with this new setting.

File: charts/partials/templates/_proxy.tpl (hunk 1 above)
Comment (mateiidavid) [reply charts/partials/templates/_proxy.tpl:163]: yeah I did not want to touch the impl but I agree.

File: charts/linkerd-control-plane/values.yaml (hunk 2 above)
Comment (mateiidavid) [reply charts/linkerd-control-plane/values.yaml:203]: it was hardcoded in to 10, see the _proxy.tpl change. The default did not change.

Comment (alpeb) [reply charts/linkerd-control-plane/values.yaml:203]: :blush:

**Comment by kflynn** [comment]
@jan-kantert Hey! We'd like to take this, if you can confirm that you agree with the whole DCO thing. A comment on this issue is fine. 🙂

**Comment by jan-kantert** [comment]
I agree to the DCO! Sorry that I did not get to it before Christmas. Thanks for taking care!

**Review by alpeb — APPROVED** [review]
I've rebased and pushed some minor changes. Good to go IMO :+1:

**Review by adleong — APPROVED** [review]
This adds a way to configure probe initialDelay and timeout.  Do we also anticipate needing to configure probe period?  Should we just add that as well while we're here?  Worth a thought but not a blocker.

**Pull request merged**